tools/cargo-miden/src/build.rs

```diff
@@ -41,6 +41,8 @@ pub fn build_masm(
         "--verbose".as_ref(),
         "-l".as_ref(),
         "std".as_ref(),
+        "-l".as_ref(),
```

Comment:
Two things:

1. We should not use `miden` as the name for any `Library`, as it is far too general and ambiguous
2. If we are going to have separate crates for each `miden-base` subcomponent (such as the TX kernel), we should be more precise with the names, e.g. `-l miden_tx_kernel` (which should then be the file stem used when writing it to disk). I add the `miden_` prefix here because if we have a number of such libraries, they are less "special", so should be namespaced to avoid conflicts
3. I would prefer to group `miden-base` libraries together, and link them using `-l base`. This would be a "special" library like `std`, which we handle in the compiler directly.
4. Rather than specifying `-l std -l base`, we should infer those options in the compiler when `--target rollup` is specified (since it is implied that those libraries are dependencies). Here, in `cargo-miden`, we should distinguish between building a crate for the rollup, and building a crate that just targets the base VM, and pass `--target` to get the default set of libraries for those targets.

I'll add target-derived default libraries today, so you can assume the default target adds `-l std`, and the `rollup` target adds both `-l std and -l  base`; but I'll leave it up to you to add handling for `-l base` in the `LinkLibrary` type of `midenc-session` like we do for `-l std`

Reply:
I don't think we need one crate for each `miden-base` subcomponent. I agree that `miden` is too generic.